src/codegen/compile.ml

```diff
@@ -2530,6 +2538,50 @@ module MakeCompact (Num : BigNumType) : BigNumType = struct
         get_buf ^^ get_x ^^ Num.compile_store_to_data_buf_signed env)
       env
 
+  let compile_store_to_stream_unsigned env =
+    let set_x, get_x = new_local env "x" in
+    let set_stream, get_stream = new_local env "stream" in
+    set_x ^^ set_stream ^^
+    get_x ^^
+    try_unbox I32Type
+      (fun env ->
+        BitTagged.untag_i32 ^^ set_x ^^
+        (* get size & reserve & encode *)
+        I32Leb.compile_store_to_data_buf_unsigned env get_x
+          begin
+            get_stream ^^
+            I32Leb.compile_leb128_size get_x ^^
+            E.call_import env "rts" "stream_reserve"
+          end)
```

Comment:
```suggestion
        let set_dst, get_dst = new_local env "dst" in
        get_stream ^^
        I32Leb.compile_leb128_size get_x ^^
        E.call_import env "rts" "stream_reserve" ^^
        set_dst  ^^
        I32Leb.compile_store_to_data_buf_unsigned env get_x get_dst)
```
Might be easier to follow (assuming equivalent). Ditto below.

Reply:
It looks equivalent (functionality-wise) but it adds two extra instructions, so I prefer to leave as-is.

Reply:
I am using an OCaml `let` instead.